core/src/main/java/com/linecorp/armeria/internal/common/PathAndQuery.java

```diff
@@ -133,7 +133,8 @@ public static PathAndQuery parse(@Nullable String rawPath) {
      * the parsed result was valid (e.g., when a server is able to successfully handle the parsed path).
      */
     public void storeInCache(@Nullable String rawPath) {
-        if (CACHE != null && rawPath != null) {
+        if (CACHE != null && rawPath != null && !cached) {
```

Comment:
Could check `cached` before checking `rawPath`? Because it's more likely to hit?